### DISCUSSION TITLE
global middleware that runs only on the client-side

### DISCUSSION THREAD

**Discussion by vhoyer** [nuxt/nuxt#9003] [Questions (Nuxt 2)]
Does anyone know a way of having something like a global middleware that runs only on the client-side (even on the first request).

I want to execute a function on the client-side :D

**Comment by TheAlexLichter** [comment]
You can use a client-side Nuxt plugin in this case 😊

**Reply by vhoyer** [reply]
But how can the plugin run something on each route? does it already run?
